docs/backend-requests/making/jwt-templates.mdx

```diff
@@ -160,7 +160,65 @@ For example, you could use interpolation to build the user's full name:
 }
 ```
 
-Interpolated shortcode substituted values will always be casted to strings. For example, if the user does not have a last name associated, the above full name value would be "null John".
+Interpolated shortcodes will always result to string values. For example, if the user does not have a last name associated, the above full name value would be "null John".
+
+### Coalescing
+
+Shortcode coalescing is a conditional expression of the form:
+
+```json
+"{{ <operand_1> || <operand_2> || <operand_n> }}"
+```
+
+For example, given the following template:
+
+```json
+{
+  "has_verified_contact_info": "{{user.has_verified_email || user.has_verified_phone}}",
+
+  // fallback to a string value
+  "full_name": "{{user.first_name || 'John'}} {{user.last_name || 'Doe' }}",
+
+  // fallback to a number value
+  "age": "{{user.public_metadata.age || user.unsafe_metadata.age || 30 }}"
+}
+```
+
+The output after substitution (subject to the actual user data) would look like this:
+
+```json
+{
+  "has_verified_contact_info": true,
+  "full_name": "Mary Doe",
+  "age": 30
+}
+```
+
+Coalescing is often used to substitute a default fallback value for shortcodes that would otherwise result in `null` or `false` values.
+
+In a coalescing expression, operands are separated by `||`. The result of a coalescing expression is the first operand that is not `null`/`false`. If all operands of the expression are `null`/`false`, the last operand is returned no matter its return value.
+
+Coalescing can be combined with interpolation:
+
+```json
+{
+  "title": "Awesome {{user.first_name || user.last_name || 'user'}}"
+}
+```
+
+Assuming a user that has no first or last name set in their profile, the above would output:
```

Comment:
❓ Do we document which shortcodes are nullable? Because for example `user.full_name` will return an empty string instead and the condition will fail to fall back to the literal value. Perhaps this could be surprising, not sure though.

Reply:
Right, this won't work as is. Changing this to a value that may indeed provide a `null` value (first/last name will always print a string).

Reply:
Changed this to use `full_name` instead, which can be `null`.